### ISSUE TITLE
[terra-dialog] Close Button Missing Required Name Property

### ISSUE THREAD

**Issue by joel-cohen** [cerner/terra-core#1397]
# Issue Description
The Terra Dialog component is missing the **name** property on the close button [here](https://github.com/cerner/terra-core/blob/master/packages/terra-dialog/src/Dialog.jsx#L43), which is required by Terra Button according to Terra Button's documentation and code.  This is the warning message currently being displayed in the browser's developer console when using a Terra Dialog:

![error](https://user-images.githubusercontent.com/37081612/38249081-76dd5860-3718-11e8-8c82-803341a45a69.png) 

## Issue Type

- [ ] New Feature
- [ ] Enhancement
- [x] Bug
- [ ] Other

## Expected Behavior
The close button in the Terra Dialog component should have a **name** property valued, which is required by Terra Button.  There then would be no warning message in the browser's developer console console.

## Current Behavior
A warning is displaying in browser's developer console indicating the required property is missing.

## Steps to Reproduce
1.  Create a component using the Terra Dialog component.
2.  Not running in production mode so warnings display, load the component with the browser's developer console open.
3.  The warning message above in the *Issue Description* section should be displayed.

## Environment
* Component Version: 1.3.0
* Browser Name and Version: Google Chrome; Version 65.0.3325.181 (Official Build) (64-bit)
* Operating System and version (desktop or mobile): macOS Sierra; Version 10.12.6

**Comment by emilyrohrbough** [comment]
Duplicate of https://github.com/cerner/terra-core/issues/1351
